Components/Model/Product.php

```diff
@@ -279,6 +284,44 @@ public function loadData(Article $article, Shop $shop = null)
     }
 
     /**
+     * update Article fields to translated text based on the shop id.
+     *
+     * @param Article $article article to be updated
+     * @param Shop|null $shop sub shop id
+     */
+    public function amendArticleTranslation(\Shopware\Models\Article\Article $article, Shop $shop = null)
+    {
+        $builder = Shopware()->Models()->createQueryBuilder();
+        $builder = $builder->select(array('translations'))
+            ->from(\Shopware\Models\Translation\Translation::class, 'translations')
+            ->where('translations.key = :articleId')->setParameter('articleId', $article->getId())
+            ->andWhere('translations.type = \'article\'');
+
+        if ($shop !== null && $shop->getId() !== null) {
+            $builder = $builder->andWhere('translations.shopId = :shopId')
+                ->setParameter('shopId', $shop->getId());
+        }
+        $query = $builder->getQuery();
+        $result = $query->getOneOrNullResult();
+
+        if ($result instanceof \Shopware\Models\Translation\Translation && $result->getData()) {
+            $dataObject = unserialize($result->getData());
+            if (array_key_exists("txtArtikel", $dataObject)) {
+                $article->setName($dataObject["txtArtikel"]);
+                $this->setName($article->getName());
+            }
+            if (array_key_exists("txtshortdescription", $dataObject)) {
+                $article->setDescription($dataObject["txtshortdescription"]);
```

Comment:
Do not alter the `$article` object. We don't want to interfere with Shopware's internal logic.